[inductor] add a few tests to verify view_to_reshape pass is safe

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103034

This PR follows up on issue https://github.com/pytorch/pytorch/issues/102229 . I added 2 unit tests and verified that autoaugorad/functionalization already handles view properly. The view_to_reshape pass does not cause an issues.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @bertmaher

shunting314 committed:
[WIP][inductor] add a few tests to verify view_to_reshape pass is safe

[ghstack-poisoned]

eellison reviewed (commented):
could you add in a Conv where we would permute to channels last ?

shunting314 commented:
> could you add in a Conv where we would permute to channels last ?

We can. I think the tests of view_to_reshape change can be independent to layout optimization. We run view_to_reshape pass with or without layout-optimization. But I can add an extra test with Conv to cover more cases.

ezyang reviewed (approved):
i'll always approve new tests like this

shunting314 committed:
Update on "[inductor] add a few tests to verify view_to_reshape pass is safe"


This PR follows up on issue https://github.com/pytorch/pytorch/issues/102229 . I added 2 unit tests and verified that autoaugorad/functionalization already handles view properly. The view_to_reshape pass does not cause an issues.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 bertmaher

[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge -f "Failed tests are unrelated. After all this PR itself just adds 4 tests"
